### PR TITLE
V3-2292 SatelliteDB create owner_id column for project table

### PR DESCRIPTION
**Pull request by crawter**
What: 
1. Column owner_id should be added to projects table
2. Projects without users will be removed
3. OwnerID column of rest projects will be filled with id of first project member

Why: 
Our new payment flow requires this changes, because only project owner is responsible for payment

Please describe the tests:
Tests updated to check ownerID field
 
Please describe the performance impact: none

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/satellitedb/migrate.go [hunk 1]
```diff
@@ -1057,6 +1057,35 @@ func (db *DB) PostgresMigration() *migrate.Migration {
...
+				Description: "Create partial index for user_credits table",
+				Version:     49,
+				Action: migrate.SQL{
+					`ALTER TABLE projects
+						ADD COLUMN owner_id BYTEA;
+
+                    DELETE FROM projects WHERE id NOT IN (
+                        SELECT project_id
+                            FROM project_members
+                                LEFT OUTER JOIN projects
```
Comment (ifraixedes) [inline satellite/satellitedb/migrate.go]: This outer join is weird here, I think that the query will be simpler doing

```sql

DELETE FROM projects WHERE id NOT IN (
  SELECT project_id
  FROM project_members
  GROUP BY (project_id)
)
```

File: satellite/satellitedb/migrate.go [hunk 2]
```diff
@@ -1057,6 +1057,35 @@ func (db *DB) PostgresMigration() *migrate.Migration {
...
+                                    ON projects.id = project_members.project_id);
+
+					UPDATE projects as proj
+                        SET owner_id = 
+                            (SELECT member_id
```
Comment (ifraixedes) [inline satellite/satellitedb/migrate.go]: If I'm not mistaken this subquery could be simpler

```sql
SELECT member_id
FROM project_members
WHERE project_id = proj.id
ORDER BY created_at ASC
LIMIT 1;
```

I've tried both doing a small simulation in http://sqlfiddle.com

Selecting Postgres 9.3
Using this schema

```sql
CREATE TABLE projects (
  id INT NOT NULL,
  UNIQUE(id)
);

INSERT INTO projects (id) VALUES (0);
INSERT INTO projects (id) VALUES (1);

CREATE TABLE members (
  member_id INT NOT NULL,
  project_id INT NOT NULL REFERENCES projects(id), 
  created_at timestamp with time zone NOT NULL,
  PRIMARY KEY (member_id, project_id)
);

INSERT INTO members (member_id, project_id, created_at) VALUES (1, 0, '2000-09-28 03:00:00');
INSERT INTO members (member_id, project_id, created_at) VALUES (1, 1, '2002-09-28 03:00:00');
INSERT INTO members (member_id, project_id, created_at) VALUES (2, 0, '2001-09-28 03:00:00');
INSERT INTO members (member_id, project_id, created_at) VALUES (2, 1, '2003-09-28 03:00:00');
```

And the following queries:

```sql
-- current subquery

SELECT member_id
FROM members
WHERE project_id = 1
GROUP BY member_id
ORDER BY MAX(created_at) ASC
LIMIT 1;


-- new subquery

SELECT member_id
FROM members
WHERE project_id = 1
ORDER BY created_at ASC
LIMIT 1;

```

File: satellite/satellitedb/migrate.go (hunk 1 above)
Comment (crawter) [reply satellite/satellitedb/migrate.go]: ure totally right

File: satellite/satellitedb/migrate.go [hunk 3]
```diff
@@ -1057,6 +1057,35 @@ func (db *DB) PostgresMigration() *migrate.Migration {
...
+				// Creating owner_id column for project.
+				// Removing projects without project members
+				// And populating this column with first project member id
+				Description: "Creating owner_id column for projects table",
+				Version:     49,
+				Action: migrate.SQL{
+					`ALTER TABLE projects
+						ADD COLUMN owner_id BYTEA;
+
+                    UPDATE projects as proj
```
Comment (egonelbre) [inline satellite/satellitedb/migrate.go]: fix identation

File: satellite/console/service.go [hunk 4]
```diff
@@ -1332,6 +1333,22 @@ func (s *Service) isProjectMember(ctx context.Context, userID uuid.UUID, project
...
+func (s *Service) isProjectOwner(ctx context.Context, userID uuid.UUID, projectID uuid.UUID) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	project, err := s.store.Projects().Get(ctx, projectID)
+	if err != nil {
+		s.log.Error("internal error", zap.Error(err))
+		return errs.New(internalErrMsg)
+	}
+
+	if project.OwnerID != userID {
+		return errs.New(unauthorizedErrMsg)
```
Comment (egonelbre) [inline satellite/console/service.go]: Let's use error classes.

**Comment by crawter** [comment]
implementation continues in https://github.com/storj/storj/pull/2706

**Pull request closed without merging**